Assets/Scripts/Core/GridCombatSystem.cs

```diff
@@ -193,5 +195,44 @@ private enum State {
 		public CoreUnit GetActiveUnit() {
 			return unitGridCombat_;
 		}
+
+		/*	
+		 *	This method calculates and returns the hit chance between two Vector3's 
+		 *	It is possible to add another float variable WeaponModifierHitChance to this method, then adjust float hitChance accordingly.
+		 */
+		private float RangedHitChance(Vector3 player, Vector3 target, float weaponMaxRange) {
+			Grid<Tilemap.Node> grid = GameController.Instance.GetGrid();
+			List<Vector3> points = CalculatePoints(player,target);
+			float hitChance = 100f;
+
+			if(weaponMaxRange >= Vector3.Distance(player, target)) {
+				// Target is in range
+				foreach(Vector3 v in points) {
+					// Calculates hitchance
+					hitChance -= grid.GetGridObject(v).hitChanceModifier;
+				}
+			}
+			return hitChance;
+		}
+
+		// The methods `Line()` and `DiagonalDistance()` are from https://www.redblobgames.com/grids/line-drawing.html and adjusted accordingly.
+
+		// Calculates the length between two Vector3's and returns N nodes between them.
+		private List<Vector3> CalculatePoints(Vector3 p0, Vector3 p1) {
+			List<Vector3> points = new List<Vector3>();
+			float diagonalLength = DiagonalDistance(p0, p1);
+			for(int step = 0; step <= diagonalLength; step++) {
+				float t = diagonalLength == 0 ? 0.0f : step / diagonalLength;
+				points.Add(Vector3.Lerp(p0, p1, t));
+			}
+			return points;
+		}
+
+		// Calculates the diagonal distance between two Vector3's.
+		private float DiagonalDistance(Vector3 v0, Vector3 v1) {
+			float diagonalX = v1.x - v0.x;
+			float diagonalY = v1.y - v0.y;
+			return Mathf.Max(Mathf.Abs(diagonalX), Mathf.Abs(diagonalY));

```

Comment:
Why is the diagonal distance between 2 vectors the maximum value of either the X difference of the Y difference. Should this not be the sqrt of the difference x squared plus the diffence y squared, or Vector3.Distance?